p2p/libp2p/directSender.go

```diff
@@ -237,15 +252,41 @@ func (ds *directSender) getOrCreateStream(conn network.Conn) (network.Stream, er
 	return foundStream, nil
 }
 
-func (ds *directSender) createMessage(topic string, buff []byte, conn network.Conn) *pubsubPb.Message {
+func (ds *directSender) createMessage(topic string, buff []byte, conn network.Conn) (*pubsubPb.Message, error) {
 	seqno := ds.NextSeqno()
 	mes := pubsubPb.Message{}
 	mes.Data = buff
 	mes.Topic = &topic
 	mes.From = []byte(conn.LocalPeer())
 	mes.Seqno = seqno
 
-	return &mes
+	buff, err := mes.Marshal()
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
in libp2p there is the `SignPrefix` that's being added, should we think about it here, while we construct the message?

Reply:
added